### ISSUE TITLE
A11y documentation audit: Review Form Controls component documentation

### ISSUE THREAD

**Issue by k80bowman** [department-of-veterans-affairs/va.gov-team#35998]
## Description
We want to audit our accessibility documentation on design.va.gov to make sure that it is up-to-date with current accessibility guidelines.

For this ticket, we need to review the [accessibility guidelines for the Form Controls components](https://design.va.gov/components/form-controls) on design.va.gov and provide any updates that need to be made to the existing documentation as a comment on this ticket.

The accessibility documentation for Form Controls exists as a sub-section for each type of control, underneath the "Accessibility considerations" heading. We expect the guidance for each form control to be reviewed.

## Details
Just for reference, here are some useful links:
[Platform accessibility standards](https://vfs.atlassian.net/wiki/spaces/pilot/pages/1777893393/WCAG+2.1+success+criteria+and+foundational+testing)
[VFS Team accessibility responsibilities](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/working-with-vsp/roles-and-responsibilities/accessibility.md#accessibility)
[VA.gov experience standards](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)
VSP WCAG 2.1 Presentations:
[VSP WCAG2.1 Foundational Testing Redux](https://app.zenhub.com/files/133843125/368da85e-7f1f-4332-8ccb-6808e52a8a51/download)
[VSP WCAG2.1 New Success Criteria](https://app.zenhub.com/files/133843125/5475fd9b-9d2e-4a6a-901f-97682664eb33/download)

## Acceptance Criteria
- [ ] Review accessibility guidelines on the following pages of design.va.gov:
    - [ ] https://design.va.gov/components/form-controls
- [ ] Make sure accessibility guidelines match the performance of the component in Storybook
- [ ] Add notes on any necessary updates as a comment on this ticket
- [ ] Ping Katy, Carol, and Natalie for review

**Comment by CourtneyChambers77** [comment]
Intext
Avoid placeholder text for accessibility reasons. Provide a hint text for supporting contextual help, this will always be shown. Hint text should sit above a form field.  Ensure hint text can be read correctly by screen readers. Most browsers’ default rendering of placeholder text does not provide a high enough contrast ratio.
Avoid breaking numbers with distinct sections (such as phone numbers, Social Security Numbers, or credit card numbers) into separate input fields. Ensure there is sufficient spacing between form elements. For example, use one input for phone number, not three (one for area code, one for local code, and one for number). Each field needs to be labeled for a screen reader and the labels for fields broken into segments are often not meaningful.
Select box
Make sure your select box has a label. Don’t replace it with the default menu option (for example, removing the “State” label and just having the dropdown read “Select a state” by default).
Don’t use JavaScript to automatically submit the form (or do anything else) when an option is selected. Auto-submission disrupts screen readers because they select each option as they read them.
Checkboxes
Use a single legend for fieldset (for required fields). A common use of the fieldset and legend elements is a question with radio-button options for answers. The question text and radio buttons are wrapped in a fieldset, with the question itself being inside the legend tag.
Surround a related set of checkboxes with a <fieldset>. The <legend> provides context for the grouping. Do not use fieldset and legend for a single check.
The custom checkboxes here are accessible to screen readers because the default checkboxes are moved off-screen with margin-left: -2rem; opacity: 0; position: absolute; left: auto.
Each input should have a semantic id attribute, and its corresponding label should have the same value in it’s for attribute.
The title attribute can replace <label>.
When using form blocks, use a vertical pattern. This approach is ideal, from an accessibility standpoint, because of limited vision that makes it hard to scan from right to left.
Allow users to submit a form. 
Ensures the data submitted by the user conforms to acceptable limits. 
Radio buttons
 
Group related radio buttons together with <fieldset> and describe the group with <legend>.
Each radio button should have a <label>. Associate the two by matching the <label>’s for attribute to the <input>’s id attribute.
The title attribute can replace <label>.
Date input
Follow text input and select box guidance. The component consists of two select boxes for month and day inputs and a text input for year input.

**Comment by k80bowman** [comment]
This looks great, thank you!
